wallet/src/de/schildbach/wallet/data/TransactionMetadataDocumentDao.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022. Dash Core Group.
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.data
+
+import androidx.room.*
+import org.bitcoinj.core.Sha256Hash
+
+/**
+ * @author Eric Britten
+ */
+@Dao
+interface TransactionMetadataDocumentDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(transactionMetadata: TransactionMetadataDocument)
+
+    @Query("SELECT * FROM transaction_metadata_platform ORDER BY timestamp")
+    suspend fun load(): List<TransactionMetadataDocument>
+
+    @Query("SELECT COUNT(*) FROM transaction_metadata_platform WHERE id = :id")
+    suspend fun count(id: String): Int
+
+    @Query("""
+        SELECT sentTimestamp 
+        FROM transaction_metadata_platform 
+        WHERE timestamp = (
+            SELECT MAX(timestamp) 
+            FROM transaction_metadata_platform 
+            WHERE txId = :txId
+                AND sentTimestamp is NOT NULL
+        )
+        """)
+    suspend fun getSentTimestamp(txId: Sha256Hash): Long?
```

Comment:
This query scheme is present on all of these get functions.

We want to obtain the most recent value of some data for a particular transaction.  This appears to have one query embedded in another.